src/streamlink/plugins/youtube.py

```diff
@@ -365,5 +419,75 @@ def _get_streams(self):
 
         return streams
 
+    def set_title_info(self):
+        if self.video_id is None:
+            self.video_id = self._find_video_id(self.url)
+
+        query = {
+            "part": "id,snippet",
+            "id": self.video_id,
+            "key": API_KEY
+        }
+        url = API_BASE + "/videos"
+        res = http.get(url, params=query, raise_for_status=False)
+        if res.status_code != codes.ok:
+            self.logger.warn(ERROR_API.format(url, res.status_code, "Title Name"))
+            return None
+        allInfo = http.json(res, schema=_title_info_schema)
+
+        for info in allInfo:
+            self.title = info["snippet"]["title"]
+            self.channel_id = info["snippet"]["channelId"]
+            self.category_id = info["snippet"]["categoryId"]
+            return
+
+    def get_title(self):
+        if self.title is None:
+            self.logger.debug("get_title")
+            self.set_title_info()
+        return self.title
+
+    def get_category(self):
+        if self.category is None:
+            self.logger.debug("get_category")
+            if self.category_id is None:
+                self.set_title_info()
+            # translate from category_id to category name
+            query = {
+                "part": "snippet",
+                "id": self.category_id,
```

Comment:
`self.category_id` could still be `None` here (same goes for `self.channel_id` in `get_author`.)

Should check for `None` and `return` if it is still `None` after the request.

Reply:
I see what you mean. Will do.